iml-agent/src/action_plugins/check_stonith.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::{CibError, ImlAgentError},
+    cmd::cmd_output,
+};
+use elementtree::Element;
+use futures::Future;
+use iml_wire_types::{ComponentState, ConfigState, ServiceState};
+use std::process::Command;
+use tracing::{span, Level};
+use tracing_futures::Instrument;
+
+fn inst_attr_problem(elem: &Option<&Element>, node: &String) -> Option<String> {
+    match elem {
```

Comment:
Have you tried using the `?` operator here? Since you are dealing with a bunch of matches / nesting, and returning an `Option`, it would dedent this code a bit.

Reply:
No, I get:
```
error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> iml-agent/src/action_plugins/check_stonith.rs:63:14
   |
63 |     for e in elem?.from_err().children() {
   |              ^^^^^ the `?` operator cannot be applied to type `&std::option::Option<&elementtree::Element>`
   |
   = help: the trait `std::ops::Try` is not implemented for `&std::option::Option<&elementtree::Element>`
   = note: required by `std::ops::Try::into_result`
```
I can flatten this more though